packages/rum/src/userActionCollection.ts

```diff
@@ -66,55 +67,110 @@ export interface AutoUserAction {
 export type UserAction = CustomUserAction | AutoUserAction
 
 export function startUserActionCollection(lifeCycle: LifeCycle) {
+  const subscriptions: Subscription[] = []
+  let currentViewId: string | undefined
+
+  addEventListener(DOM_EVENT.CLICK, processClick, { capture: true })
   function processClick(event: Event) {
     if (!(event.target instanceof Element)) {
       return
     }
-
     newUserAction(lifeCycle, UserActionType.CLICK, getElementContent(event.target))
   }
 
-  addEventListener(DOM_EVENT.CLICK, processClick, { capture: true })
+  subscriptions.push(lifeCycle.subscribe(LifeCycleEventType.VIEW_COLLECTED, processViewLoading))
+  function processViewLoading(loadedView: View) {
+    if (!loadedView || loadedView.id === currentViewId) {
+      return
+    }
+    currentViewId = loadedView.id
+    newViewLoading(lifeCycle, loadedView.location.pathname, loadedView.startTime)
+  }
 
   return {
     stop() {
       removeEventListener(DOM_EVENT.CLICK, processClick, { capture: true })
+      subscriptions.forEach((s) => s.unsubscribe())
     },
   }
 }
 
-let currentUserAction: { id: string; startTime: number } | undefined
+interface PartialUserAction {
+  id: string
+  startTime: number
+  type: UserActionType
+  name: string
+}
+let currentUserAction: PartialUserAction | undefined
+
+interface ViewLoadingState {
+  pathname: string
+  startTime: number
+  stopPageActivitiesTracking: () => void
+}
+let currentViewLoadingState: ViewLoadingState | undefined
+
+function newViewLoading(lifeCycle: LifeCycle, pathname: string, startTime: number) {
+  // Cancel current user action
+  currentUserAction = undefined
+
+  if (currentViewLoadingState) {
+    currentViewLoadingState.stopPageActivitiesTracking()
+  }
+
+  const { observable: pageActivitiesObservable, stop: stopPageActivitiesTracking } = trackPageActivities(lifeCycle)
+  currentViewLoadingState = {
+    pathname,
+    startTime,
+    stopPageActivitiesTracking,
+  }
+
+  waitUserActionCompletion(pageActivitiesObservable, (endTime) => {
```

Comment:
since we are not using that for user action, we should rename this function and the used constants to something more appropriate.
like wait waitPageActivitiesEnded, PAGE_ACTIVITY_VALIDATION_DELAY, ...